tests/unit/test_utils.py

```diff
@@ -207,6 +209,135 @@ def test_fail(self, x: torch.Tensor):
         assert not utils.is_lower_triangular(x)
 
 
+@pytest.mark.parametrize(
+    "prepend_space_to_answer, expected_print_calls, expected_rprint_calls",
```

Comment:
I think you are checking that it tokenises correctly here? 

```suggestion
    "prepend_space_to_answer, tokenized_prompt, tokeized_answer",
```

I'd suggest changing to this ^ and then just putting the array in directly. The `call()` part can be done in the test.